### PR TITLE
qemu sec gpio

### PR DESCRIPTION
**Pull request by muvarov**
update qemu to mailine 5.2 and add secure gpio for reset/poweroff.

### CODE REVIEW

File: meta-ledge-bsp/recipes-devtools/qemu/qemu.inc [hunk 1]
```diff
@@ -11,31 +11,20 @@ RDEPENDS_${PN}-ptest = "bash make"
...
-           file://0005-qemu-native-set-ld.bfd-fix-cflags-and-set-some-envir.patch \
-           file://0006-chardev-connect-socket-to-a-spawned-command.patch \
-           file://0007-apic-fixup-fallthrough-to-PIC.patch \
-           file://0008-linux-user-Fix-webkitgtk-hangs-on-32-bit-x86-target.patch \
-           file://0009-Fix-webkitgtk-builds.patch \
-           file://0010-configure-Add-pkg-config-handling-for-libgcrypt.patch \
-           file://0001-Add-enable-disable-udev.patch \
-	   file://0001-qemu-Do-not-include-file-if-not-exists.patch \
-	   file://CVE-2020-13361.patch \
-	   file://find_datadir.patch \
```
Comment (apalos) [inline meta-ledge-bsp/recipes-devtools/qemu/qemu.inc:33]: All older patches are included in 5.2?

File: meta-ledge-bsp/recipes-bsp/trusted-firmware-a/trusted-firmware-a-ledge_git.bb [hunk 2]
```diff
@@ -16,6 +16,8 @@ PV = "2.3"
 SRC_URI = "git://github.com/ARM-software/arm-trusted-firmware.git;protocol=https;nobranch=1"
 SRCREV = "fc721f830855ab05f45d8f2024325601bbb4c783"
 
+SRC_URI += "file://0001-plat-qemu-trigger-reboot-with-secure-pl061.patch"
+
```
Comment (apalos) [inline meta-ledge-bsp/recipes-bsp/trusted-firmware-a/trusted-firmware-a-ledge_git.bb:20]: I assume this hasn't been merged upstream yet?

**Review by apalos — CHANGES_REQUESTED** [review]

File: meta-ledge-bsp/recipes-devtools/qemu/qemu.inc [hunk 3]
```diff
@@ -11,31 +11,20 @@ RDEPENDS_${PN}-ptest = "bash make"
...
+       file://0001-hw-gpio-implement-gpio-pwr-driver-for-qemu-reset-pow.patch \
+       file://0002-arm-virt-refactor-gpios-creation.patch \
+       file://0003-arm-virt-add-secure-pl061-for-reset-power-down.patch \
 	   "
```
Comment (apalos) [inline meta-ledge-bsp/recipes-devtools/qemu/qemu.inc:23]: Are the qemu patrches merged in upstream? So this is a temporary patch until we get the next qemu version?

**Review by apalos — COMMENTED** [review]
From the commit message it's not clear why this is needed or if it's part of mainline QEMU.

If you reboot patches have been accepted in upstream, can you add something on the commit message that explains why we need those? I assume the patches are temporary until we bump our QEMU release?

File: meta-ledge-bsp/recipes-devtools/qemu/qemu.inc (hunk 3 above)
Comment (muvarov) [reply meta-ledge-bsp/recipes-devtools/qemu/qemu.inc:23]: Peter staged them to his qemu-arm branch. They have to be merged to qemu 6.0

**Comment by muvarov** [comment]
Yes, this is temporary patch until qemu will release 6.0. This will take some time and we need fixed reboot now.

**Review by apalos — APPROVED** [review]

**Comment by muvarov** [comment]
Merged with squash fix patch to the main patch.

**Pull request closed without merging**